jaeger-influxdb/internal/influxdb_reader.go

```diff
@@ -54,8 +61,12 @@ func (ir *influxdbReader) GetTrace(ctx context.Context, traceID model.TraceID) (
 		return nil
 	}
 	err = executeQuery(ctx, ir.db, queryGetTraceEvents(ir.tableLogs, traceID), f)
+	if isTableNotFound(err) {
+		// ignore; the table hasn't been written to yet, so it doesn't exist (schema-on-write)
+		err = nil
+	}
 	if err != nil {
-		ir.logger.Info("ignoring query error", zap.String("table", ir.tableLogs), zap.Error(err))
+		return nil, err
```

Comment:
Might be a bit less fiddling with the value of `err` for these cases to use a `switch` this way:

```go
switch {
    case isTableNotFound(err):
        // ignore; the table hasn't been written to yet, so it doesn't exist (schema-on-write)
    case err != nil:
        return nil, err
}
```

I find this kind of error switching a little bit more readable, but YMMV!

Reply:
I dig it! After refactoring that way, here's what I settled on in most cases:

```golang
if err != nil && !isTableNotFound(err) { // ignore table not found (schema-on-write)
	return nil, err
}
```

And once I did this:
```golang
switch {
case err != nil && !isTableNotFound(err): // ignore table not found (schema-on-write)
	return nil, err
case len(spansBySpanID) == 0:
	return nil, spanstore.ErrTraceNotFound
}
```